austation/code/game/gamemodes/bloodsucker/bloodsucker.dm

```diff
@@ -0,0 +1,304 @@
+
+/datum/game_mode
+	var/list/datum/mind/bloodsuckers = list() 		// List of minds belonging to this game mode.
+	var/list/datum/mind/vassals = list() 			// List of minds that have been turned into Vassals.
+	var/list/datum/mind/vamphunters = list() 		// List of minds hunting vampires.
+	var/obj/effect/sunlight/bloodsucker_sunlight	// Sunlight Timer. Created on first Bloodsucker assign. Destroyed on last removed Bloodsucker.
+
+	// LISTS //
+	var/list/vassal_allowed_antags = list(/datum/antagonist/brother, /datum/antagonist/traitor, /datum/antagonist/traitor/internal_affairs, /datum/antagonist/survivalist, \
+										  /datum/antagonist/rev, /datum/antagonist/nukeop, /datum/antagonist/pirate, /datum/antagonist/cult, /datum/antagonist/abductee)
+	// The antags you're allowed to be if turning Vassal.
+
+/datum/game_mode/bloodsucker
+	name = "bloodsucker"
+	config_tag = "bloodsucker"
+	report_type = "traitor"
+	traitor_name = "Bloodsucker"//Nanotrasen Internal Affairs Agent"
+	antag_flag = ROLE_BLOODSUCKER
+	false_report_weight = 1
+	restricted_jobs = list("AI","Cyborg")
+	protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Head of Personnel", "Brig Physician")
+	required_players = 0
+	required_enemies = 1
+	recommended_enemies = 4
+	reroll_friendly = 1
+	enemy_minimum_age = 7
+
+	announce_span = "danger"
+	announce_text = "Filthy, bloodsucking vampires are crawling around disguised as crewmembers!\n\
+	<span class='danger'>Bloodsuckers</span>: The crew are cattle, while you are both shepherd and slaughterhouse.\n\
+	<span class='notice'>Crew</span>: Put an end to the undead infestation before the station is overcome!"
+
+
+
+// Seems to be run by game ONCE, and finds all potential players to be antag.
+/datum/game_mode/bloodsucker/pre_setup()
+
+	// Set Restricted Jobs
+	if(CONFIG_GET(flag/protect_roles_from_antagonist))
+		restricted_jobs += protected_jobs
+
+	if(CONFIG_GET(flag/protect_assistant_from_antagonist))
+		restricted_jobs += "Assistant"
+
+	// Set number of Vamps
+	recommended_enemies = clamp(round(num_players()/10), 1, 6);
+
+	// Select Antags
+	if(antag_candidates.len>0)
+		for(var/i = 0, i < recommended_enemies, i++)
+			if (!antag_candidates.len)
+				break
+			var/datum/mind/bloodsucker = antag_pick(antag_candidates, ROLE_BLOODSUCKER)
+			//Can we even BE a bloodsucker?
+			if (can_make_bloodsucker(bloodsucker, display_warning=FALSE))
+				bloodsuckers += bloodsucker
+				bloodsucker.special_role = ROLE_BLOODSUCKER
+				bloodsucker.restricted_roles = restricted_jobs
+				log_game("[bloodsucker.key] (ckey) has been selected as a Bloodsucker.")
+				antag_candidates.Remove(bloodsucker) // Apparently you can also write antag_cndidates -= bloodsucker
+		return 1
```

Comment:
```suggestion
				i-- //because fuck you
			antag_candidates.Remove(bloodsucker) // Apparently you can also write antag_cndidates -= bloodsucker
		if(bloodsuckers.len >= required_enemies)
			return 1
		else
			setup_error = "Not enough VALID bloodsucker candidates"
			return 0
```

This fixes multiple issues in bloodsucker selection but it really is a meh one. It works, and it doesn't _have_ to be fixed, but FIXME!!